File: src/main/java/com/qtech/bubbles/gui/QPanel.java
package com.qtech.bubbles.gui;

import com.qtech.bubbles.common.GraphicsProcessor;

public class QPanel extends QContainer {
    @Override
    public void renderComponents(GraphicsProcessor ngg) {
        components.forEach(c -> c.render(ngg));
    }

    @Override
    public void render(GraphicsProcessor ngg) {
        GraphicsProcessor ngg2 = ngg.create(getX(), getY(), getWidth(), getHeight());
        renderComponent(ngg2);
        ngg2.dispose();

        GraphicsProcessor ngg3 = ngg.create(getX(), getY(), getWidth(), getHeight());
        renderComponents(ngg3);
        ngg3.dispose();
    }

    @Override
    public void renderComponent(GraphicsProcessor ngg) {
        ngg.setColor(getBackgroundColor());
        ngg.fill(getBounds());
    }

    @Override
    public void tick() {

    }
}

File: src/main/kotlin/qtech/bubbles/gui/QPanel.kt
package qtech.bubbles.gui

import qtech.hydro.GraphicsProcessor
import java.util.function.Consumer

class QPanel : QContainer() {
    override fun renderComponents(ngg: GraphicsProcessor) {
        components!!.forEach(Consumer { c: QComponent -> c.render(ngg) })
    }

    override fun render(ngg: GraphicsProcessor) {
        val ngg2 = ngg.create(x, y, width, height)
        renderComponent(ngg2)
        ngg2.dispose()
        val ngg3 = ngg.create(x, y, width, height)
        renderComponents(ngg3)
        ngg3.dispose()
    }

    override fun renderComponent(ngg: GraphicsProcessor) {
        ngg.color(backgroundColor)
        ngg.shapeF(bounds)
    }

    override fun tick() {}
}